server/webapp/WEB-INF/rails.new/webpack/single_page_apps/artifact_stores.js

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const $                    = require('jquery');
+const m                    = require('mithril');
+const ArtifactStoresWidget = require('views/artifact_stores/artifact_stores_widget');
+require('foundation-sites');
+
+$(() => {
+  $(document).foundation();
+
+  m.mount($("#artifact-stores").get(0), ArtifactStoresWidget);
```

Comment:
Shouldn't you make a plugin info call to find out whether there are plugins of type `artifact` installed? Check `single_page_apps/elastic_profiles.js`.

Reply:
I've done this in artifact_stores_widget so that it can be tested. It remains untested (and difficult to test) in elastic_profiles